doc/source/data/random-access.rst

```diff
@@ -0,0 +1,49 @@
+---------------------------------
+Random Data Access (Experimental)
+---------------------------------
+
+Any Arrow-format dataset can be enabled for random access by calling ``dataset.to_random_access_dataset(key="col_name")``. This partitions the dataset across the cluster by the given sort key, providing efficient random access to records via binary search. A number of worker actors are created, each of which has zero-copy access to the underlying sorted data blocks of the Dataset.
```

Comment:
Technically pandas format is implementation detail (users cannot distinguish between Arrow and pandas format dataset, unless you call the private _dataset_format method).

Reply:
Got it. Actually I think that implementation detail can be easily leaked? For example arrow and pandas support different data types.

Reply:
I think the semantics we want us that Arrow is the canonical format. So things can temporarily be in pandas for efficiency reasons, but eventually it will Arrowified.